src/rebar_app_discover.erl

```diff
@@ -164,7 +164,12 @@ create_app_info(AppDir, AppFile) ->
             AppInfo1 = rebar_app_info:applications(
                          rebar_app_info:app_details(AppInfo, AppDetails),
                          IncludedApplications++Applications),
-            Valid = rebar_app_utils:validate_application_info(AppInfo1),
+            Valid = case rebar_app_utils:validate_application_info(AppInfo1) of
+                        true ->
+                            true;
+                        _ ->
```

Comment:
It is also used when it is required to be valid and an error message along with the reason it isn't valid is returned to the user: https://github.com/rebar/rebar3/blob/master/src/rebar_otp_app.erl#L86